make debian-packages won't build packages

hello,

On debian 7.8 wheezy, or ubuntu 14.04.2 LTS trusty, ```make debian-packages``` does not build the packages. dpkg-source complains about some upstream files beeing modified... Here is how to reproduce :
```
git clone https://github.com/pdav/netmagis.git
cd netmagis
make debian-packages
```
And the error message : 
```
dpkg-source: info: using source format `3.0 (quilt)'
dpkg-source: info: building netmagis using existing ./netmagis_2.2.3.orig.tar.gz
dpkg-source: info: local changes detected, the modified files are:
 netmagis-2.2.3/Makefile
 netmagis-2.2.3/common/netmagis.conf.sample
dpkg-source: error: aborting due to unexpected upstream changes, see /tmp/netmagis_2.2.3-1.diff.nS1Xax
dpkg-source: info: you can integrate the local changes with dpkg-source --commit
dpkg-buildpackage: error: dpkg-source -b netmagis-2.2.3 gave error exit status 2
debuild: fatal error at line 1364:
dpkg-buildpackage -rfakeroot -D -us -uc failed
make[1]: *** [release] Error 29
make[1]: Leaving directory `/home/schplurtz/SOFT/netmagis/pkg/debian'
make: *** [debian-packages] Error 2
```

--- a/pkg/debian/Makefile
+++ b/pkg/debian/Makefile
@@ -10,10 +10,22 @@
 
 release: clean
 	@echo ">>> Making release in `pwd`"
-	@which debuild >/dev/null 2>&1 || \
-	    ( echo 1>&2 "You need the 'devscripts' package: apt-get -y install devscripts" && exit 1 )
-	@which dh_make >/dev/null 2>&1 || \
-	    ( echo 1>&2 "You need the 'dh-make' package: apt-get -y install dh-make" && exit 2 )
+	@missing_pkgs= ;                                                \
+	 for cmd_pkg_pair in                                            \
+	   "quilt         quilt"                                        \
+	   "debuild       devscripts"                                   \
+	   "dh_make       dh-make" ;                                    \
+	 do                                                             \
+	   set -- $$cmd_pkg_pair ;                                      \
+	   which $$1 >/dev/null 2>&1 ||                                 \
+	     missing_pkgs="$$missing_pkgs $$2" ;                        \
+	 done ;                                                         \
+	 test "$$missing_pkgs" && {                                     \
+	   printf '%s\n' "missing packages : $$missing_pkgs"            \
+	                 "you may install them with :"                  \
+	                 "apt-get -y install$$missing_pkgs" ;           \
+	   exit 1 ;                                                     \
+	 } || : yeah. We have all the needed packages...
 	sh ./gendeb $(VERSION)
 
 release-arch:

--- a/pkg/debian/gendeb
+++ b/pkg/debian/gendeb
@@ -1,5 +1,7 @@
 #!/bin/sh
 
+set -e # Stop on first error
+
 # Variables
 export DEBEMAIL=jean@unistra.fr
 export DEBFULLNAME="Jean BENOIT"
@@ -17,6 +19,10 @@ VERSION=$1
 
 TGZ=netmagis-$VERSION.tar.gz
 URL=$BASEURL/$TGZ
+QUILTRC=$PWD/quiltrc
+dquilt() {
+  quilt --quiltrc="$QUILTRC" "$@"
+}
 
 PORTSDIR=$PWD/../freebsd
 
@@ -103,14 +109,20 @@ rm -fr $DIR
 
 tar xzf $TGZ
 
-modmakefile $DIR/Makefile
-
-modconf $DIR/common/netmagis.conf.sample
-
 cd $DIR
 
 # Prepare debian directory and backup orig file
 dh_make -m -p netmagis_${VERSION} -f ../$TGZ </dev/null
+dquilt new install.patch
+dquilt add Makefile
+dquilt add common/netmagis.conf.sample
+
+(
+cd ..
+modmakefile $DIR/Makefile
+
+modconf $DIR/common/netmagis.conf.sample
+)
 
 # Remove unused templates
 rm -f debian/*.EX debian/*.ex
@@ -138,4 +150,13 @@ etc/init.d/topographd
 etc/init.d/toposendd
 EOF
 
+dquilt refresh
+dquilt header
+# set patch headers as per http://dep.debian.net/deps/dep3/
+dquilt header -r <<EOH
+Description: Changes needed to build deb packages
+ Path adjustments inside Makefile and example file.
+Author: $DEBFULLNAME <$DEBEMAIL>
+Forwarded: not-needed
+EOH
 debuild -us -uc

--- a/pkg/debian/quiltrc
+++ b/pkg/debian/quiltrc
@@ -1,0 +1,12 @@
+# see https://www.debian.org/doc/manuals/maint-guide/modify.fr.html
+d=. ; while [ ! -d $d/debian -a `readlink -e $d` != / ]; do d=$d/..; done
+if [ -d $d/debian ] && [ -z $QUILT_PATCHES ]; then
+    # dans le cas d'une arborescence de paquet Debian avec $QUILT_PATCHES non configuré
+    QUILT_PATCHES="debian/patches"
+    QUILT_PATCH_OPTS="--reject-format=unified"
+    QUILT_DIFF_ARGS="-p ab --no-timestamps --no-index --color=auto"
+    QUILT_REFRESH_ARGS="-p ab --no-timestamps --no-index"
+    QUILT_COLORS="diff_hdr=1;32:diff_add=1;34:diff_rem=1;31:diff_hunk=1;33:diff_ctx=35:diff_cctx=33"
+    if ! [ -d $d/debian/patches ]; then mkdir $d/debian/patches; fi
+fi
+

